### PR TITLE
8328005: Convert java/awt/im/JTextFieldTest.java applet test to main

### PR DESCRIPTION
**Pull request by DamonGuy**
Convert java/awt/im/JTextFieldTest.java applet test to main using PassFailJFrame.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8328005](https://bugs.openjdk.org/browse/JDK-8328005): Convert java/awt/im/JTextFieldTest.java applet test to main (**Bug** - P4)

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**) ⚠️ Review applies to [c27b6139](https://git.openjdk.org/jdk/pull/18247/files/c27b61391f17a94e0ca7897e91ceb0474addc90d)
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - Committer) ⚠️ Review applies to [c27b6139](https://git.openjdk.org/jdk/pull/18247/files/c27b61391f17a94e0ca7897e91ceb0474addc90d)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18247/head:pull/18247` \
`$ git checkout pull/18247`

Update a local copy of the PR: \
`$ git checkout pull/18247` \
`$ git pull https://git.openjdk.org/jdk.git pull/18247/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18247`

View PR using the GUI difftool: \
`$ git pr show -t 18247`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18247.diff">https://git.openjdk.org/jdk/pull/18247.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18247#issuecomment-1992337247)

### CODE REVIEW

File: test/jdk/java/awt/im/JTextFieldTest.java [hunk 1]
```diff
@@ -23,43 +23,55 @@
...
+    public static JFrame createAndShowGUI() {
+        frame = new JFrame("Test Frame");
+        frame.setLayout(new FlowLayout());
+        TextField tf1 = new TextField("ABCDEFGH", 10);
+        tf1.setEditable(false);
+        JTextField tf2 = new JTextField("12345678", 10);
+        frame.getContentPane().add(tf1);
+        frame.getContentPane().add(tf2);
+        frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
+        frame.setVisible(true);
```
Comment (honkar-jdk) [inline test/jdk/java/awt/im/JTextFieldTest.java:73]: frame.setVisible(true) not required here. Can be removed has PassFailJFrame manages it.

File: test/jdk/java/awt/im/JTextFieldTest.java [hunk 2]
```diff
@@ -23,43 +23,55 @@
...
-        add(tf2);
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame
+            .builder()
+            .title("JTextFieldTest")
+            .instructions(INSTRUCTIONS)
+            .rows(15)
+            .columns(40)
+            .position(PassFailJFrame.Position.TOP_LEFT_CORNER)
+            .testUI(JTextFieldTest::createAndShowGUI)
```
Comment (honkar-jdk) [inline test/jdk/java/awt/im/JTextFieldTest.java:59]: TestUI automatically positioned .position() is redundant

```suggestion
            .testUI(JTextFieldTest::createAndShowGUI)
```

File: test/jdk/java/awt/im/JTextFieldTest.java [hunk 3]
```diff
@@ -23,43 +23,53 @@
...
+import javax.swing.WindowConstants;
 
-    TextField  tf1;
-    JTextField tf2;
+public class JTextFieldTest {
+    private static JFrame frame;
+    private static final String INSTRUCTIONS =
+            """
+             Please run this test case under Solaris cjk locale with inputmethod
+             support, if you could input Chinese/Japanese/Korean in the swing
```
Comment (prrace) [inline test/jdk/java/awt/im/JTextFieldTest.java:47]: The scenario here should not be specific to Solaris.
I think you should rephrase the first sentence to
"Please run this test in a CJK (Chinese/Japanese/Korean) locale with input method support".

I some times wonder how well these tests that say "please change your environment" actually get properly tested.
It is perhaps possible to do change desktop locale support on the fly but by the time
you read the instructions the test is already running.
That introduces another wrinkle. Maybe you need to restart at least this test.
And I am not 100% sure that jtreg passes through locale vars. I hope it does.

Comment (DamonGuy) [reply test/jdk/java/awt/im/JTextFieldTest.java]: Updated the instructions.

File: test/jdk/java/awt/im/JTextFieldTest.java [hunk 4]
```diff
@@ -23,43 +23,53 @@
...
-        win.show();
+    public static JFrame createAndShowGUI() {
+        frame = new JFrame("Test Frame");
+        frame.setLayout(new FlowLayout());
+        TextField tf1 = new TextField("ABCDEFGH", 10);
+        tf1.setEditable(false);
+        JTextField tf2 = new JTextField("12345678", 10);
+        frame.getContentPane().add(tf1);
+        frame.getContentPane().add(tf2);
+        frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
```
Comment (kumarabhi006) [inline test/jdk/java/awt/im/JTextFieldTest.java:71]: frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
Redundat, can be removed.

Comment (DamonGuy) [reply test/jdk/java/awt/im/JTextFieldTest.java]: Removed. Thanks!

**Review by prrace — APPROVED** [review]

File: test/jdk/java/awt/im/JTextFieldTest.java [hunk 5]
```diff
@@ -23,43 +23,51 @@
...
- * @bug  4226191
+ * @bug 4226191
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
  * @summary  Verify that Lightweight text components (like swing JTextField)
  *           work correctly with IM when there is an uneditable peered
  *           TextField/TextArea in the same parent Frame
- * @author xueming.shen@eng
- * @run applet/manual=yesno JTextFieldTest.html
+ * @run main JTextFieldTest
```
Comment (kumarabhi006) [inline test/jdk/java/awt/im/JTextFieldTest.java:32]: should be main/manual ?

File: test/jdk/java/awt/im/JTextFieldTest.java [hunk 6]
```diff
@@ -23,43 +23,51 @@
...
+import java.awt.TextField;
 
-public class JTextFieldTest extends Applet implements ActionListener {
+import javax.swing.JFrame;
+import javax.swing.JTextField;
 
-    TextField  tf1;
-    JTextField tf2;
+public class JTextFieldTest {
+    private static JFrame frame;
```
Comment (kumarabhi006) [inline test/jdk/java/awt/im/JTextFieldTest.java:42]: can be moved to local scope in `createAndShowGUI` method.

**Review by kumarabhi006 — APPROVED** [review]

File: test/jdk/java/awt/im/JTextFieldTest.java (hunk 5 above)
Comment (DamonGuy) [reply test/jdk/java/awt/im/JTextFieldTest.java]: Fixed.

File: test/jdk/java/awt/im/JTextFieldTest.java (hunk 6 above)
Comment (DamonGuy) [reply test/jdk/java/awt/im/JTextFieldTest.java]: Fixed.

**Pull request closed without merging**